crates/ty_python_semantic/src/types/subclass_of.rs

```diff
@@ -249,6 +336,7 @@ impl<'db> VarianceInferable<'db> for SubclassOfType<'db> {
 pub(crate) enum SubclassOfInner<'db> {
     Class(ClassType<'db>),
     Dynamic(DynamicType),
+    TypeVar(BoundTypeVarInstance<'db>),
 }
```

Comment:
An alternative design might be to just have it be its own `Type` variant (`Type::MetaTypeVar`?). That might save you from a few of your `.unwrap()` calls in `Type::has_relation_to_impl`, for example -- there are quite a few places where you need to treat this `SubclassOfInner` variant differently to all the other ones.

I don't have a strong opinion, though. That design would probably bring its own complications.